src/master-playlist-controller.js

```diff
@@ -817,7 +817,7 @@ export class MasterPlaylistController extends videojs.EventTarget {
     // trying to load the master OR while we were disposing of the tech
     if (!currentPlaylist) {
       this.error = error;
-      return this.mediaSource.endOfStream('network');
+      return this.trigger('error');
```

Comment:
What about putting the `endOfStream` call in a try catch and if it fails trigger the error on MPC? I feel like we probably want to still call `endOfStream` in the common case of not hitting an `InvalidStateError`